source/slang/slang-ir-liveness.cpp

```diff
@@ -256,6 +265,10 @@ struct LivenessContext
         /// Get the block from the index
     IRBlock* _getBlock(BlockIndex blockIndex) const { return m_functionBlockInfos[Index(blockIndex)].block; }
 
+        /// True if the terminator can be considered an access
+        /// This allows us to elide a scope end if the root is returned 
+    bool _isAccessTerminator(IRTerminatorInst* terminator);
+
```

Comment:
We should double-check the semantic rules for the relevant SPIR-V ops. The LLVM live-range instructions say that something like:

When a thread "executes" a live-range start, the contents of the relevant memory become undefined (so it needs to be emitted before any stores to initialize it), and the memory is considered live until a matching live-range-end instruction is executed **or** the function returns.

That second bit about function return is important, because it would imply that we never need to emit *any* live-range-end operations before a `return` (whether or not the live value is being returned).